.golangci.yml

```diff
@@ -126,3 +126,37 @@ linters-settings:
           - github.com/DataDog/datadog-agent/pkg/util/log.Printf
           - github.com/DataDog/datadog-agent/pkg/util/log.Warnf
           - github.com/DataDog/datadog-agent/pkg/util/log.Errorf
+  revive:
+    # in order to change revive config, the default configuration must be explicitly defined
+    # https://github.com/mgechev/revive/blob/master/defaults.toml
+    ignoreGeneratedHeader: false
+    severity: "warning"
+    confidence: 0.8
+    errorCode: 0
+    warningCode: 0
+    rules:
+      - name: blank-imports
+      - name: context-as-argument
+      - name: context-keys-type
+      - name: dot-imports
+      - name: empty-block
+      - name: error-naming
+      - name: error-return
+      - name: error-strings
+      - name: errorf
+      - name: exported
+      - name: increment-decrement
+      - name: indent-error-flow
+      - name: package-comments
+      - name: range
+      - name: receiver-naming
+      - name: redefines-builtin-id
+      - name: superfluous-else
+      - name: time-naming
+      - name: unexported-return
+      - name: unreachable-code
+      - name: unused-parameter
+      - name: var-declaration
+      - name: var-naming
+      # non-default rules:
+      - name: duplicated-imports
```

Comment:
This config is the one from `revive` master right ? Maybe we should use the one from the version of `revive` our current version of `golangci-lint` is using instead ?

Reply:
Nice catch !
I checked, and the only difference between the config from master and the one from 1.3.2 (the version we are using) is that they dropped the `if-return` rule ([PR](https://github.com/mgechev/revive/pull/843)) which shouldn't have been part of the default (it was accidentally added back).
I think we should keep it out of our rules.